compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -1377,7 +1370,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
 
             let slot = self.get_personality_slot(&mut bx);
             slot.storage_live(&mut bx);
```

Comment:
The personality slot could be split up to store both elements separately. That would save on field projecrions.

Reply:
Ah, you meant at the internal Rust level -- sure, yes, that seems possible to change. I can try that out.

Reply:
Hm -- @bjorn3 my worry is that if we split up the landing pad slot such that it's two separate allocas, then we are largely back to the current situation where LLVM IR is somewhat verbose to unpack the result of a landingpad instruction (which returns the aggregate in LLVM IR) and then re-pack it when invoking resume later. It doesn't seem like it would really carry ~any performance impact, but I'm not sure it's worth making this refactoring given the need to unpack and repack at least with LLVM.

I also took a look at whether we could avoid the unpacking and then repacking by having a more linear control flow (i.e., presuming that each landing pad is eventually followed by a resume, so we can just use the SSA variable result of landingpad instruction directly), but it seems difficult to make that change with today's compiler at least (though I suspect it might be a good idea; I'm not sure it's worth it though).

I think my sense is that we should just close this PR at this point and chalk it up as not really helpful.

Reply:
> It doesn't seem like it would really carry ~any performance impact

I wouldn't expect it to given that the resume instruction turns into a tail call to `_Unwind_Resume` with both struct fields as separate arguments in separate registers. The landing pad also gets it passed in using two separate registers.

Reply:
Sure, there's no runtime impact -- this PR is mostly about our codegen of LLVM IR rather than runtime code.

It seems to me that if we make cleanup_landing_pad return `[Self::Value; 2]` that's still going to force us to generate unpacking and repacking code with LLVM, since it's expecting an aggregate rather than distinct functions. It seems like what we almost want is a backend-specific type here, since some backends want it as an aggregate and some don't. But that seems way overkill, which is why I'm increasingly leaning towards doing nothing here.